tests/emerald/emerald_test.go

```diff
@@ -0,0 +1,192 @@
+
+	// Create the snapshot using a high level of isolation; we don't want another
+	// tx to be able to modify the tables while this is running, creating a snapshot that
+	// represents indexer state at two (or more) blockchain heights.
+	if err := target.SendBatchWithOptions(ctx, batch, pgx.TxOptions{IsoLevel: pgx.Serializable}); err != nil {
+		return 0, err
+	}
+
+	var checkpointHeight uint64
+	if err := target.QueryRow(ctx, fmt.Sprintf(`
+		SELECT height from %s.checkpointed_heights
+			WHERE analyzer='%s'
+			ORDER BY height DESC LIMIT 1;
+	`, chainID, EmeraldName)).Scan(&checkpointHeight); err != nil {
+		return 0, err
+	}
+
+	return checkpointHeight, nil
+}
+
+func TestEmeraldAccounts(t *testing.T) {
+	if _, ok := os.LookupEnv("OASIS_INDEXER_HEALTHCHECK"); !ok {
+		t.Skip("skipping test since healthcheck tests are not enabled")
+	}
+
+	t.Log("Initializing data stores...")
+
+	ctx := context.Background()
+
+	factory, err := newSourceClientFactory()
+	require.Nil(t, err)
+
+	oasisConsensusClient, err := factory.Consensus()
+	require.Nil(t, err)
+
+	network, err := analyzer.FromChainContext(MainnetChainContext)
+	require.Nil(t, err)
+
+	id, err := analyzer.RuntimeEmerald.ID(network)
+	require.Nil(t, err)
+	t.Log("Runtime ID determined", "runtime", analyzer.RuntimeEmerald.String(), "runtime_id", id)
+
+	oasisRuntimeClient, err := factory.Runtime(id)
+	require.Nil(t, err)
+
+	postgresClient, err := newTargetClient(t)
+	assert.Nil(t, err)
+
+	t.Log("Creating checkpoint for emerald tables...")
+	height, err := checkpointRuntimeBackends(t, oasisRuntimeClient, oasisConsensusClient, postgresClient)
+	assert.Nil(t, err)
+
+	t.Logf("Fetching accounts information at height %d...", height)
+	addresses, err := oasisRuntimeClient.GetAccountAddresses(ctx, height, sdkTypes.NativeDenomination)
+	assert.Nil(t, err)
+	expectedAccts := make(map[sdkTypes.Address]bool)
+	for _, addr := range addresses.Addresses {
+		expectedAccts[addr] = true
+	}
+	t.Logf("Fetched %d account addresses", len(addresses.Addresses))
+
+	acctRows, err := postgresClient.Query(ctx, fmt.Sprintf(
+		`SELECT account_address, balance
+			FROM %s.runtime_sdk_balances_checkpoint`, chainID),
+	)
+	require.Nil(t, err)
+	actualAccts := make(map[string]bool)
+	for acctRows.Next() {
+		var a TestRuntimeAccount
+		err = acctRows.Scan(
+			&a.Address,
+			&a.Balance,
+		)
+		assert.Nil(t, err)
+
+		// Check that the account exists.
+		var actualAddr sdkTypes.Address
+		err = actualAddr.UnmarshalText([]byte(a.Address))
+		assert.Nil(t, err)
+		_, ok := expectedAccts[actualAddr]
+		if !ok {
+			t.Logf("address %s found, but not expected", a.Address)
+			t.Fail()
+			continue
+		}
+
+		// Check that the account balance is accurate.
+		balances, err := oasisRuntimeClient.GetAccountBalances(ctx, height, actualAddr)
+		assert.Nil(t, err)
+		foundNativeBalance := false
+		for denom, amount := range balances.Balances.Balances {
+			if !denom.IsNative() {
+				continue
+			}
+			foundNativeBalance = true
```

Comment:
It's a pity to go to extra trouble to exclude everything but the native balance. (And we don't care for the "no native balance found" check; that's testing the node, not the indexer.) It's true that today's runtimes only have one denomination, but let's just test them all? It should make the code simpler if anything.